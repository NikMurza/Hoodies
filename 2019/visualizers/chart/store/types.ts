import { ChartData } from '@common/api/types/chart';

export interface ChartVisualizerInstance {
  openId: string;
  data?: ChartData;
  isFetching: boolean;
}

export interface ChartVisualizersState {
  [instanceId: string]: ChartVisualizerInstance;
}

export interface StatePart {
  chartVisualizers?: ChartVisualizersState;
}
export type GetState = () => StatePart;

export interface ChartPayload<T> {
  instanceId: string;
  data?: T;
}
