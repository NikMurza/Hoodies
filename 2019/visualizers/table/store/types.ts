import { PPSom } from '@common/api/types/PPSom';

export interface TableVisualizerInstance {
  openId: string;
  tabSheetData: TabSheetRawData;
  isFetching: boolean;
}

export interface TableVisualizersState {
  [instanceId: string]: TableVisualizerInstance;
}

export interface StatePart {
  tableVisualizers?: TableVisualizersState;
}
export type GetState = () => StatePart;

export interface TabSheetPayload<T> {
  instanceId: string;
  data?: T;
}

export interface TabSheetRawData {
  /** Структура и диапазоны данных. */
  Data?: PPSom.TabData[];
  /** Загруженные диапазоны. */
  Ranges?: PPSom.TabRange[];
  /** Метаданные */
  Metadata?: PPSom.TabMd;
  /** Выделение (Дублируется из метаданных). */
  Selection?: PPSom.TabSelection;
}
