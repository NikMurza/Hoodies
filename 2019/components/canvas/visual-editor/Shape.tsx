import * as React from 'react';
import { mapTypeToComponent } from '@common/components/lib/canvas/shapes';
import {
  VisualEditorActions,
  VisualEditorContextMenuIdentity,
  VisualEditorShape,
  VisualEditorSize,
  VisualEditorComponentData,
} from '@common/components/lib/types/visual-editor';
import { typeContextMenuId } from '@common/components/lib/canvas/components/context-menu/with-context-menu/WithContextMenu';

interface Props {
  item: VisualEditorShape;
  contextMenuId: string;
  stageSize: VisualEditorSize;
  hoveredShape: { id: string; type: string };
  onShapeAction: <T extends VisualEditorContextMenuIdentity>(
    action: VisualEditorActions,
    data: T,
  ) => void;
  onShapeAddComponent: (data: VisualEditorComponentData) => void;
  onShapeRemoveComponent: (id: string) => void;
}

const Shape: React.FC<Props> = ({
  item,
  contextMenuId,
  stageSize,
  hoveredShape,
  onShapeAction,
  onShapeAddComponent,
  onShapeRemoveComponent,
}) => {
  const { id, type } = item;
  const ShapeComponent = mapTypeToComponent[type];
  const isContextMenuShow = contextMenuId === typeContextMenuId(id, type);
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const props = item as any;

  // перехватываем onAction у shape
  const onAction = React.useCallback(
    <T extends VisualEditorContextMenuIdentity>(action: VisualEditorActions, data: T): void => {
      onShapeAction(action, data);
      if (props.onAction) props.onAction(action, data);
    },
    [props, onShapeAction],
  );
  return (
    <ShapeComponent
      key={`${type}_${id}`}
      {...props}
      onAction={onAction}
      isHovered={id === hoveredShape.id && type === hoveredShape.type}
      isContextMenuShow={isContextMenuShow}
      stageSize={stageSize}
      onAddComponent={onShapeAddComponent}
      onRemoveComponent={onShapeRemoveComponent}
    />
  );
};

export default Shape;
