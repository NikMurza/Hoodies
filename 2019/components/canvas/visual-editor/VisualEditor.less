.visual-editor {
  position: relative;
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;

  & .visual-canvas {
    position: relative;
    overflow: hidden;
    outline: none;

    &-scrollable {
      position: relative;
      flex-grow: 1;
      overflow: auto;
      &_hidden-scroll {
        -ms-overflow-style: none;
        scrollbar-width: none;
      }
      &_hidden-scroll::-webkit-scrollbar {
        display: none;
      }
    }
    &__components {
      position: absolute;
      top: 0;
      left: 0;
    }
    &__context-menu-wrapper {
      position: absolute;
    }
  }
}
