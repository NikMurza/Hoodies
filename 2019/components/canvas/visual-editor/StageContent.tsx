import * as React from 'react';
import { Store } from 'redux';
import { Provider } from 'react-redux';
import * as ReactKonva from 'react-konva';

import {
  VisualEditorActions,
  VisualEditorContextMenuIdentity,
  VisualEditorShape,
  VisualEditorSize,
  VisualEditorComponentData,
} from '@common/components/lib/types/visual-editor';
import Shape from './Shape';

export interface Props {
  store?: Store;
  items?: VisualEditorShape[];
  hoveredShape: { id: string; type: string };
  contextMenuId: string;
  stageSize: VisualEditorSize;
  onShapeAction: <T extends VisualEditorContextMenuIdentity>(
    action: VisualEditorActions,
    data: T,
  ) => void;
  onShapeAddComponent: (data: VisualEditorComponentData) => void;
  onShapeRemoveComponent: (id: string) => void;
}

const StageContent: React.FC<Props> = ({
  store,
  items,
  hoveredShape,
  contextMenuId,
  stageSize,
  onShapeAction,
  onShapeAddComponent,
  onShapeRemoveComponent,
}) => {
  const stage = React.useMemo(() => {
    const isHoveredShape = (shape: VisualEditorShape): boolean =>
      shape.id === hoveredShape.id && shape.type === hoveredShape.type;
    const ordered = items.filter((item) => !isHoveredShape(item));
    ordered.push(...items.filter(isHoveredShape));

    const renderItem = (item: VisualEditorShape): JSX.Element =>
      React.createElement(Shape, {
        key: `${item.type}-${item.id}`,
        item,
        contextMenuId,
        stageSize,
        hoveredShape,
        onShapeAction,
        onShapeAddComponent,
        onShapeRemoveComponent,
      });

    return <ReactKonva.Layer>{ordered.map(renderItem)}</ReactKonva.Layer>;
  }, [
    items,
    hoveredShape,
    contextMenuId,
    stageSize,
    onShapeAction,
    onShapeAddComponent,
    onShapeRemoveComponent,
  ]);

  return store ? <Provider store={store}>{stage}</Provider> : stage;
};

export default StageContent;
