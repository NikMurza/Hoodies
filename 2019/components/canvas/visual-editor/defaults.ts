import colorsConst from '@common/styles/constants/colors';
import fontConst from '@common/styles/constants/fonts';
import { VisualEditorSize } from '@common/components/lib/types/visual-editor';

interface ScaleDefaults {
  initial: number;
  min: number;
  max: number;
  step: number;
}
export interface CanvasDefaults {
  size: VisualEditorSize;
  gridCell: number;
  gridLineWidth: number;
  canvasOffset: number;
  zoom: ScaleDefaults;
}

// измерения холста
export const canvas: CanvasDefaults = {
  size: {
    width: window.innerWidth,
    height: window.innerHeight,
  },
  gridCell: 96, // сторона ячейки сетки
  gridLineWidth: 1,
  canvasOffset: 24, // минимально допустимое расстояние от края холста до элемента,
  zoom: {
    initial: 1,
    min: 0.1,
    max: 4,
    step: 0.1,
  },
};

// размер элемента по-умолчанию
export const element = {
  width: 200,
  height: 100,
  indent: 100, // размер отступа (по-умолчанию) между элементами
  radius: 60, // Радиус элемента (по-умолчанию)
  borderRadius: 0,
  borderWidth: 1,
  referencePointRadius: 6,
  editInputWidth: 100,
  editInputHeight: 24,
};

export const line = {
  borderWidth: 2,
  hoverWidth: 20,
  endpointLength: 11.5,
  endpointWidth: 10,
  iconRadius: 16,
};

export const font = {
  family: fontConst.family,
  textSize: 14,
  accentTextSize: 16,
  secondaryTextSize: 13,
};

export const colors = {
  backgroundColor: colorsConst.white,
  transparent: 'transparent',
  gridLinesColor: '#ECEDEF',
  shadowColor: '#cccccc',
  canvas: '#F2F3F4',
  circleColor: '#9DAFED',
  circleBackgroundColor: '#A469AD',
  circleBorderColor: '#8B5A94',
  circleActiveBackgroundColor: '#3B5EDC',
  circleActiveBorderColor: '#FFFFFF',
  subTextColor: '#93989E',
  sourceTypeColor: colorsConst.green,
  destTypeColor: '#59BDF8',
  convertorTypeColor: '#FFC700',
  blockShadowColor: 'rgba(115,115,115,0.12)',
  blockActiveColor: colorsConst.blue,
  borderColor: colorsConst.darkAdditional,
  referencePointsColor: colorsConst.white,
  referencePointsHoverColor: colorsConst.blue,
  referencePointsBorderColor: colorsConst.blue,
  iconSettingBackgroundColor: colorsConst.white,
  iconSettingColor: colorsConst.white,
};
