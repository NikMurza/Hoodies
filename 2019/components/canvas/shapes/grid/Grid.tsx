import Konva from 'konva';
import * as React from 'react';
import * as ReactKonva from 'react-konva';
import * as defaults from '@common/components/lib/canvas/visual-editor/defaults';
import {
  VisualEditorGrid,
  VisualEditorPosition,
  VisualEditorShapeType,
  VisualEditorSize,
} from '@common/components/lib/types/visual-editor';

export const DEFAULT_POSITION_X = 0;
export const DEFAULT_POSITION_Y = 0;
export const DEFAULT_CANVAS_SIZE = defaults.canvas.size;
export const DEFAULT_GRID_COLOR = defaults.colors.gridLinesColor;
export const DEFAULT_GRID_CELL_SIZE = defaults.canvas.gridCell;

const Grid: React.FC<VisualEditorGrid> = (props: VisualEditorGrid): JSX.Element => {
  const { position, coveredArea, color, cellSize, lineWidth, stageSize } = props;
  const { x, y } = position;
  const width = coveredArea.width ?? stageSize.width;
  const height = coveredArea.height ?? stageSize.height;
  const horizontalLinesCount = Math.ceil(height / cellSize);
  const verticalLinesCount = Math.ceil(width / cellSize);
  const lines = [];

  const onContextMenu = (e: Konva.KonvaEventObject<PointerEvent>): void => {
    e.evt.preventDefault();
  };

  // начинаем с 1, чтобы не рисовать линии у самого края холста
  for (let i = 1; i < horizontalLinesCount; i++) {
    lines.push(
      <ReactKonva.Line
        key={`hz_${i}`}
        points={[x, cellSize * i + y, width + x, cellSize * i + y]}
        stroke={color}
        strokeWidth={lineWidth}
        onContextMenu={onContextMenu}
      />,
    );
  }
  for (let i = 1; i < verticalLinesCount; i++) {
    lines.push(
      <ReactKonva.Line
        key={`vt_${i}`}
        points={[cellSize * i + x, y, cellSize * i + x, height + y]}
        stroke={color}
        strokeWidth={lineWidth}
        onContextMenu={onContextMenu}
      />,
    );
  }

  return <>{lines.map((line) => line)}</>;
};

interface CreateGridOpts {
  id?: string;
  position?: VisualEditorPosition;
  coveredArea?: VisualEditorSize;
  color?: string;
  cellSize?: number;
  lineWidth?: number;
}

/**
 * Создание элемента "Сетка" со всеми параметрами
 * @param {CreateGridOpts}  opts Опции
 */
export const createGrid = (opts?: CreateGridOpts): VisualEditorGrid => {
  opts = opts || {};
  const { id, position, coveredArea, color, cellSize, lineWidth } = opts;
  return {
    id: id || '',
    type: VisualEditorShapeType.Grid,
    position: {
      x: position ? position.x : DEFAULT_POSITION_X,
      y: position ? position.y : DEFAULT_POSITION_Y,
    },
    coveredArea: coveredArea || DEFAULT_CANVAS_SIZE,
    color: color || DEFAULT_GRID_COLOR,
    cellSize: cellSize || DEFAULT_GRID_CELL_SIZE,
    lineWidth: lineWidth || 1,
  };
};

Grid.defaultProps = {
  ...createGrid(),
};

export default React.memo(Grid);
