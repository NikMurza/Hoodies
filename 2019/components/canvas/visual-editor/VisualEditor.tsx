import * as React from 'react';

import { ReactReduxContext } from 'react-redux';
import Konva from 'konva';
import * as ReactKonva from 'react-konva';

import './VisualEditor.less';
import {
  VisualEditorActionCallback,
  VisualEditorActions,
  VisualEditorContextMenuIdentity,
  VisualEditorIdentityPosition,
  VisualEditorKeyboardAction,
  VisualEditorPosition,
  VisualEditorPositionalClick,
  VisualEditorShape,
  VisualEditorShapeWithRef,
  VisualEditorSize,
} from '@common/components/lib/types/visual-editor';
import { MouseButtons } from '@common/components/lib/types/mouseEvent';
import * as defaults from '@common/components/lib/canvas/visual-editor/defaults';
import { usePosition } from '@common/components/lib/canvas/hooks/usePosition';
import { useIsStage } from '@common/components/lib/canvas/hooks/useIsStage';
import WithDrop from '@common/components/lib/canvas/visual-editor/with-drop/WithDrop';
import WithResizeDetect, {
  WithResizeable,
} from '@common/components/lib/canvas/visual-editor/with-resize-detect/WithResizeDetect';
import StageContent from './StageContent';
import useAddRemoveComponents from './use-add-remove-component/useAddRemoveComponent';

export interface VELayer extends Konva.LayerConfig {
  children: JSX.Element;
}

export interface Props {
  /**
   * Идентификатор конкретного экземпляра VE
   */
  id?: string;

  /**
   * Элементы для отрисовки
   */
  items?: VisualEditorShape[];

  /*
   * Дополнительные слои ниже основного с контентом
   * Рекомендуется в них отображать дополнительные данные — grid, FPS и пр
   * См. useFPS, useGrid, useZoom
   */
  extraLayers?: JSX.Element;

  /*
   * Положение Stage
   * Значение по умолчанию - { x: 0, y: 0 }
   */
  position?: VisualEditorPosition;

  /*
   * Масштабирование содержимого Visual Editor, указывается в долях от 0,1 и выше.
   * Значение по умолчанию - 1, что соответствует 100%
   */
  zoom?: number;

  /*
   * Контролируем возможность перемещаться по канвасу
   * По умолчанию true
   */
  panEnabled?: boolean;

  /**
   * Вызывается на каких-либо событиях (изменение, удаление, контекстное меню)
   * @param action Событие
   * @param data Данные
   */
  onAction?: VisualEditorActionCallback;
}

const VisualEditorContainer = WithDrop(
  WithResizeDetect(
    (
      props: React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement> &
        WithResizeable,
    ): JSX.Element => {
      const { resizeableRef, ...rest } = props;
      return <div ref={resizeableRef} {...rest} />;
    },
  ),
);

/**
 * Визуальный редактор - отображает таблицы со связями (и не только)
 * @param props Свойства
 */
const VisualEditor: React.FC<React.PropsWithChildren<Props>> = ({
  id,
  position,
  zoom: customZoom = 1,
  items,
  extraLayers,
  children,
  panEnabled,
  onAction,
}) => {
  const [contextMenuId, setContextMenuId] = React.useState<string>('');
  const [hoveredShape, setHoveredShape] = React.useState({ id: '', type: '' });

  const [stageSize, setStageSize] = React.useState<VisualEditorSize>({
    width: window.innerWidth,
    height: window.innerHeight,
  });

  const stagePosition: VisualEditorPosition = {
    x: position?.x ?? 0,
    y: position?.y ?? 0,
  };

  const zoom = customZoom ?? defaults.canvas.zoom.initial;

  const stageRef = React.useRef<ReactKonva.Stage>(null);
  const isStage = useIsStage<Event>();

  const { components, onShapeAddComponent, onShapeRemoveComponent } = useAddRemoveComponents();

  const hideContextMenu = React.useCallback((): void => {
    setContextMenuId('');
  }, []);

  React.useEffect(() => {
    onAction<VisualEditorShapeWithRef<ReactKonva.Stage>>(VisualEditorActions.Mount, {
      forwardRef: stageRef,
    });
  }, [stageRef]);

  // Закрываем контекстное меню при правом клике на пустую область,
  // восстанавливаем действие по умолчанию.
  const onStageContextMenu = React.useCallback(
    (e: Konva.KonvaEventObject<PointerEvent>): void => {
      if (isStage(e)) {
        e.evt.preventDefault();
        const { x, y } = usePosition<PointerEvent>(e);

        onAction<VisualEditorPositionalClick>(VisualEditorActions.ContextMenu, {
          id,
          x,
          y,
          target: e.target,
          event: { ...e?.evt, button: MouseButtons.Right },
        });
      }
    },
    [hideContextMenu, onAction],
  );

  const onStageClick = React.useCallback(
    (e: Konva.KonvaEventObject<PointerEvent>): void => {
      const { x, y } = usePosition<PointerEvent>(e);
      onAction<VisualEditorPositionalClick>(VisualEditorActions.Click, {
        id,
        x,
        y,
        event: e?.evt,
        target: e?.target,
      });
      hideContextMenu();
    },
    [id, onAction],
  );

  const onStageMouseAction = React.useCallback(
    (action: VisualEditorActions) => (e: Konva.KonvaEventObject<MouseEvent>): void => {
      const { x, y } = usePosition<MouseEvent>(e);
      onAction<VisualEditorIdentityPosition>(action, {
        id,
        x,
        y,
      });
    },
    [id, onAction],
  );

  const onStageMouseMove = React.useCallback(onStageMouseAction(VisualEditorActions.MouseMove), [
    stageRef,
    onStageMouseAction,
  ]);

  const onStageMouseUp = React.useCallback(onStageMouseAction(VisualEditorActions.MouseUp), [
    stageRef,
    onStageMouseAction,
  ]);

  const onKeyboardAction = React.useCallback(
    (key: VisualEditorActions, event: React.KeyboardEvent<HTMLDivElement>): void => {
      onAction<VisualEditorKeyboardAction>(key, {
        id,
        isCtrl: event.ctrlKey,
        isShift: event.shiftKey,
        isAltKey: event.altKey,
        keyCode: event.keyCode,
        key: event.key,
        isRepeat: event.repeat,
      });
    },
    [id, onAction],
  );

  const onStageKeyDown = React.useCallback(
    (event: React.KeyboardEvent<HTMLDivElement>): void =>
      onKeyboardAction(VisualEditorActions.KeyDown, event),
    [onKeyboardAction],
  );

  const onStageKeyUp = React.useCallback(
    (event: React.KeyboardEvent<HTMLDivElement>): void =>
      onKeyboardAction(VisualEditorActions.KeyUp, event),
    [onKeyboardAction],
  );

  const onStageDrag = React.useCallback(
    (e: Konva.KonvaEventObject<DragEvent>) => {
      if (!isStage(e)) {
        return;
      }

      onAction<VisualEditorIdentityPosition>(VisualEditorActions.Move, {
        id,
        x: -e.target.x(),
        y: -e.target.y(),
      });
    },
    [id, onAction],
  );

  const onStageResize = React.useCallback(
    (size: VisualEditorSize): void => {
      setStageSize(size);
      onAction<VisualEditorSize>(VisualEditorActions.Resize, size);
    },
    [onAction],
  );

  const onShapeAction = React.useCallback(
    <T extends VisualEditorContextMenuIdentity>(action: VisualEditorActions, data: T): void => {
      const actions = {
        [VisualEditorActions.ContextMenu]: (data: VisualEditorContextMenuIdentity): void =>
          setContextMenuId(data.contextMenuId),
        [VisualEditorActions.Move]: hideContextMenu,
        [VisualEditorActions.Hover]: (data: VisualEditorShape): void => {
          setHoveredShape({ id: data.id, type: data.type });
        },
      };

      if (actions[action]) actions[action](data);
    },
    [hideContextMenu],
  );

  React.useEffect(() => {
    const selector = document.getElementById('root');
    if (!selector) {
      return null;
    }
    selector.addEventListener('click', hideContextMenu);
    selector.addEventListener('contextmenu', hideContextMenu, true);
    return (): void => {
      selector.removeEventListener('click', hideContextMenu);
      selector.removeEventListener('contextmenu', hideContextMenu);
    };
  }, []);

  const renderStage: boolean = stageSize.width > 0 && stageSize.height > 0;

  return (
    <VisualEditorContainer
      onResize={onStageResize}
      className="visual-editor"
      stageRef={stageRef}
      onAction={onAction}
    >
      <div
        tabIndex={-1}
        className="visual-canvas"
        onKeyDown={onStageKeyDown}
        onKeyUp={onStageKeyUp}
      >
        <ReactReduxContext.Consumer>
          {({ store }): JSX.Element =>
            !renderStage ? null : (
              <ReactKonva.Stage
                draggable={panEnabled}
                ref={stageRef}
                id={id}
                x={-stagePosition.x}
                y={-stagePosition.y}
                width={stageSize.width}
                height={stageSize.height}
                scaleX={zoom}
                scaleY={zoom}
                stageRef={stageRef}
                onClick={onStageClick}
                onContextMenu={onStageContextMenu}
                onMouseMove={onStageMouseMove}
                onMouseUp={onStageMouseUp}
                onDragEnd={onStageDrag}
              >
                {extraLayers}
                {/* подключаем к стору конву, если задан store */}
                {React.createElement(StageContent, {
                  store,
                  items,
                  hoveredShape,
                  contextMenuId,
                  stageSize,
                  onShapeAction,
                  onShapeAddComponent,
                  onShapeRemoveComponent,
                })}
              </ReactKonva.Stage>
            )
          }
        </ReactReduxContext.Consumer>
        {/* отрисовываем дополнительные компоненты */}
        <div className="visual-canvas__components">
          {Object.values(components)}
          {/* отрисовываем детей */}
          {children}
        </div>
      </div>
    </VisualEditorContainer>
  );
};

// по-умолчанию редактор занимает размер экрана
VisualEditor.defaultProps = {
  id: 'VisualEditor',
  items: [],
  panEnabled: true,
  onAction: (): void => undefined,
};

export default VisualEditor;
