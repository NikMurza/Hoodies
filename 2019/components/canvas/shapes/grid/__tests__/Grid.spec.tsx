import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import { VisualEditorShapeType } from '@common/components/lib/types/visual-editor';
import Grid, {
  createGrid,
  DEFAULT_POSITION_X,
  DEFAULT_POSITION_Y,
  DEFAULT_CANVAS_SIZE,
  DEFAULT_GRID_COLOR,
  DEFAULT_GRID_CELL_SIZE,
} from '../Grid';

describe('Grid component', () => {
  it('renders correctly', () => {
    const coveredArea = {
      width: 500,
      height: 500,
    };
    const color = '#ccc';
    const cellSize = 50;
    const lineWidth = 1;
    expect(
      toJson(
        shallow(
          <Grid
            type={VisualEditorShapeType.Grid}
            id="grid"
            coveredArea={coveredArea}
            color={color}
            cellSize={cellSize}
            lineWidth={lineWidth}
          />,
        ),
      ),
    ).toMatchSnapshot();
  });

  it('createCircle works correctly with default props', () => {
    expect(createGrid()).toEqual({
      id: '',
      type: VisualEditorShapeType.Grid,
      position: {
        x: DEFAULT_POSITION_X,
        y: DEFAULT_POSITION_Y,
      },
      coveredArea: DEFAULT_CANVAS_SIZE,
      color: DEFAULT_GRID_COLOR,
      cellSize: DEFAULT_GRID_CELL_SIZE,
      lineWidth: 1,
    });
  });

  it('createCircle works correctly with custom props', () => {
    const id = 'table1';
    expect(
      createGrid({
        id: 'testId',
        position: {
          x: 100,
          y: 2000,
        },
        coveredArea: {
          width: 100000,
          height: 200000,
        },
        cellSize: 10,
      }),
    ).toEqual({
      id: 'testId',
      type: VisualEditorShapeType.Grid,
      position: {
        x: 100,
        y: 2000,
      },
      coveredArea: {
        width: 100000,
        height: 200000,
      },
      color: DEFAULT_GRID_COLOR,
      cellSize: 10,
      lineWidth: 1,
    });
  });
});
