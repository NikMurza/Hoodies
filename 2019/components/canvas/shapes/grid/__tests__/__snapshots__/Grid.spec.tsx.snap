// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component renders correctly 1`] = `
<Fragment>
  <Line
    key="hz_1"
    onContextMenu={[Function]}
    points={
      Array [
        0,
        50,
        500,
        50,
      ]
    }
    stroke="#ccc"
    strokeWidth={1}
  />
  <Line
    key="hz_2"
    onContextMenu={[Function]}
    points={
      Array [
        0,
        100,
        500,
        100,
      ]
    }
    stroke="#ccc"
    strokeWidth={1}
  />
  <Line
    key="hz_3"
    onContextMenu={[Function]}
    points={
      Array [
        0,
        150,
        500,
        150,
      ]
    }
    stroke="#ccc"
    strokeWidth={1}
  />
  <Line
    key="hz_4"
    onContextMenu={[Function]}
    points={
      Array [
        0,
        200,
        500,
        200,
      ]
    }
    stroke="#ccc"
    strokeWidth={1}
  />
  <Line
    key="hz_5"
    onContextMenu={[Function]}
    points={
      Array [
        0,
        250,
        500,
        250,
      ]
    }
    stroke="#ccc"
    strokeWidth={1}
  />
  <Line
    key="hz_6"
    onContextMenu={[Function]}
    points={
      Array [
        0,
        300,
        500,
        300,
      ]
    }
    stroke="#ccc"
    strokeWidth={1}
  />
  <Line
    key="hz_7"
    onContextMenu={[Function]}
    points={
      Array [
        0,
        350,
        500,
        350,
      ]
    }
    stroke="#ccc"
    strokeWidth={1}
  />
  <Line
    key="hz_8"
    onContextMenu={[Function]}
    points={
      Array [
        0,
        400,
        500,
        400,
      ]
    }
    stroke="#ccc"
    strokeWidth={1}
  />
  <Line
    key="hz_9"
    onContextMenu={[Function]}
    points={
      Array [
        0,
        450,
        500,
        450,
      ]
    }
    stroke="#ccc"
    strokeWidth={1}
  />
  <Line
    key="vt_1"
    onContextMenu={[Function]}
    points={
      Array [
        50,
        0,
        50,
        500,
      ]
    }
    stroke="#ccc"
    strokeWidth={1}
  />
  <Line
    key="vt_2"
    onContextMenu={[Function]}
    points={
      Array [
        100,
        0,
        100,
        500,
      ]
    }
    stroke="#ccc"
    strokeWidth={1}
  />
  <Line
    key="vt_3"
    onContextMenu={[Function]}
    points={
      Array [
        150,
        0,
        150,
        500,
      ]
    }
    stroke="#ccc"
    strokeWidth={1}
  />
  <Line
    key="vt_4"
    onContextMenu={[Function]}
    points={
      Array [
        200,
        0,
        200,
        500,
      ]
    }
    stroke="#ccc"
    strokeWidth={1}
  />
  <Line
    key="vt_5"
    onContextMenu={[Function]}
    points={
      Array [
        250,
        0,
        250,
        500,
      ]
    }
    stroke="#ccc"
    strokeWidth={1}
  />
  <Line
    key="vt_6"
    onContextMenu={[Function]}
    points={
      Array [
        300,
        0,
        300,
        500,
      ]
    }
    stroke="#ccc"
    strokeWidth={1}
  />
  <Line
    key="vt_7"
    onContextMenu={[Function]}
    points={
      Array [
        350,
        0,
        350,
        500,
      ]
    }
    stroke="#ccc"
    strokeWidth={1}
  />
  <Line
    key="vt_8"
    onContextMenu={[Function]}
    points={
      Array [
        400,
        0,
        400,
        500,
      ]
    }
    stroke="#ccc"
    strokeWidth={1}
  />
  <Line
    key="vt_9"
    onContextMenu={[Function]}
    points={
      Array [
        450,
        0,
        450,
        500,
      ]
    }
    stroke="#ccc"
    strokeWidth={1}
  />
</Fragment>
`;
